From b1bd2f406eab321be642decd6aee6b6222aec62b Mon Sep 17 00:00:00 2001
From: Dave Stevenson <dave.stevenson@raspberrypi.com>
Date: Fri, 28 Jul 2023 17:40:27 +0100
Subject: [PATCH] drm/panel: simple: Alter the timing for the Pi 7" DSI display

vc4 has always fixed up the timing, so the values defined have
never actually appeared on the wire.
The display appears to want a slightly longer HFP, so extend
the timings and recompute the clock to give the same frame rate.

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
---
 drivers/gpu/drm/panel/panel-simple.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

--- a/drivers/gpu/drm/panel/panel-simple.c
+++ b/drivers/gpu/drm/panel/panel-simple.c
@@ -3241,11 +3241,11 @@ static const struct panel_desc qishenglo
 };
 
 static const struct drm_display_mode raspberrypi_7inch_mode = {
-	.clock = 25979400 / 1000,
+	.clock = 27777,
 	.hdisplay = 800,
-	.hsync_start = 800 + 2,
-	.hsync_end = 800 + 2 + 2,
-	.htotal = 800 + 2 + 2 + 46,
+	.hsync_start = 800 + 59,
+	.hsync_end = 800 + 59 + 2,
+	.htotal = 800 + 59 + 2 + 46,
 	.vdisplay = 480,
 	.vsync_start = 480 + 7,
 	.vsync_end = 480 + 7 + 2,
